src/main/java/nl/tudelft/jpacman/board/Board.java

```diff
@@ -23,7 +23,7 @@
      */
     Board(Square[][] grid) {
         assert grid != null;
-        this.board = grid;
+        this.board = (Square[][]) grid.clone();
```

Comment:
why are we using clone and a cast here?

Reply:
Well spotted ;-). 

This relates to the PMD rule [Array is stored directly](https://pmd.github.io/pmd-6.2.0/pmd_rules_java_bestpractices.html#arrayisstoreddirectly).

The deeper issue is that we should rethink the pmd rule set, conform #121.

Reply:
I think this will run into more problems than it would solve. If we do want to address it, we must make sure that we can actually copy a Square properly. That said, if we need to keep this in to keep PMD happy let's add a TODO to make sure we address this as soon as possible (once #121 is resolved)

Reply:
Yes. "Solving" this non-problem would require a deep copy, which Arrays.copyOf also does not do.

My proposal is to undo the clone addition, add a SuppressWarnings here referencing the rule, and then revisit this when we recheck the pmd rule set to be used. (And I'm beginning to be in favor of dropping this rule).

Reply:
@avandeursen Yes that seems to be an appropriate solution 👍